crates/slot-arithmetic/src/lib.rs

```diff
@@ -74,11 +85,113 @@ impl<'b, C> Decode<'b, C> for Slot {
     }
 }
 
+#[derive(Clone, Debug, Copy, PartialEq, PartialOrd, Ord, Eq, Serialize, Deserialize, Default)]
+#[repr(transparent)]
+pub struct Epoch(u64);
+
+#[cfg(any(test, feature = "test-utils"))]
+impl Arbitrary for Epoch {
+    type Parameters = ();
+    type Strategy = BoxedStrategy<Self>;
+
+    fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
+        (0..u64::MAX).prop_map(Epoch::from).boxed()
+    }
+}
+
+impl Display for Epoch {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl FromStr for Epoch {
+    type Err = std::num::ParseIntError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        s.parse::<u64>().map(Epoch)
+    }
+}
+
+impl From<u64> for Epoch {
+    fn from(epoch: u64) -> Epoch {
+        Epoch(epoch)
+    }
+}
+
+impl From<Epoch> for u64 {
+    fn from(epoch: Epoch) -> u64 {
+        epoch.0
+    }
+}
+
+impl<C> Encode<C> for Epoch {
+    fn encode<W: minicbor::encode::Write>(
+        &self,
+        e: &mut minicbor::Encoder<W>,
+        ctx: &mut C,
+    ) -> Result<(), minicbor::encode::Error<W::Error>> {
+        self.0.encode(e, ctx)
+    }
+}
+
+impl<'b, C> Decode<'b, C> for Epoch {
+    fn decode(d: &mut Decoder<'b>, _ctx: &mut C) -> Result<Self, minicbor::decode::Error> {
+        d.u64().map(Epoch)
+    }
+}
+
+impl Add<u64> for Epoch {
```

Comment:
I think we discussed "domain types" support for arithmetic operations when I implemented this for `Slot` and I vaguely remember we decided to not do this, but can't remember the specifics. I think the argument was that it was better to provided dedicated method where it made sense.